### ISSUE TITLE
Pub get tar permission denied

### ISSUE THREAD

**Issue by bergwerf** [dart-lang/pub#1967]
I get this puzzling error when trying to download the dependencies for http://github.com/bergwerf/fltk-dart

```
IO  : HTTP response 200 OK for GET https://pub.dartlang.org/packages/image/versions/2.0.4.tar.gz
    | took 0:00:00.735075
    | x-goog-storage-class: STANDARD
    | last-modified: Thu, 14 Jun 2018 04:25:45 GMT
    | cache-control: public, max-age=3600
    | date: Mon, 13 Aug 2018 14:10:31 GMT
    | x-goog-stored-content-length: 7086984
    | x-guploader-uploadid: AEnB2UqJceDg7ab_Mr-0GJAt1v4r-JPDYHO0mN2rj7K-6ZidxeQHw8eW-MddeNfWqGZg0DykNWfQV0CvSPTPDDyMJgSgKCw2SA
    | x-goog-hash: crc32c=Ba3V+A==,md5=P9yPGwP1+OQU7oTfjyx/0w==
    | content-type: application/octet-stream
    | x-goog-generation: 1528950345602956
    | server: UploadServer
    | alt-svc: quic=":443"; ma=2592000; v="44,43,39,35"
    | accept-ranges: bytes
    | content-length: 7086984
    | x-goog-stored-content-encoding: identity
    | etag: "3fdc8f1b03f5f8e414ee84df8f2c7fd3"
    | x-goog-metageneration: 2
    | expires: Mon, 13 Aug 2018 15:10:31 GMT
FINE: Extracting .tar.gz stream to /home/herman/.pub-cache/_temp/dirANZAJW.
ERR : Permission denied
    |   Command: tar --version
FINE: Exception type: ProcessException
FINE: dart:io                                    Process.runSync
    | package:pub/src/io.dart 890:24             _computeNoUnknownKeyword
    | package:pub/src/io.dart 887:32             _noUnknownKeyword
    | package:pub/src/io.dart 887:12             _noUnknownKeyword
    | package:pub/src/io.dart 857:7              extractTarGz
    | ===== asynchronous gap ===========================
    | dart:async                                 _AsyncAwaitCompleter.completeError
    | package:pub/src/io.dart                    extractTarGz
    | ===== asynchronous gap ===========================
    | dart:async                                 _asyncThenWrapperHelper
    | package:pub/src/source/hosted.dart         BoundHostedSource._download
    | package:pub/src/source/hosted.dart 199:13  BoundHostedSource.downloadToSystemCache
    | ===== asynchronous gap ===========================
    | dart:async                                 _asyncThenWrapperHelper
    | package:pub/src/entrypoint.dart            Entrypoint.acquireDependencies
    | package:pub/src/command/get.dart 38:23     GetCommand.run
    | package:args/command_runner.dart 194:27    CommandRunner.runCommand
```

**Comment by natebosch** [comment]
What OS are you on?

Does the `tar` command work for you in a shell?

**Comment by bergwerf** [comment]
Yes it works in shell. I use debian stretch.
This project uses dependencies that contain native Dart VM extensions, perhaps something with permissions is off. I think you should be able to reproduce it on another machine.

**Comment by natebosch** [comment]
Yes, I am able to repro. Will take a look

**Comment by bergwerf** [comment]
Ok. I checked in the FLTK source code so I would have to put less effort into finding it in the future (although I am pretty sure http://www.fltk.org/index.php will keep existing for the foreseeable future). There are now also some issues with packages that do not support Dart 2.0 (cairodart, which was a Cairo binding for Dart but which is no longer maintained), so I will see if I can get to patching that.

This was just a cool experiment a while ago that I was trying to rebuild under Dart 2.0 ;).

**Comment by sortie** [comment]
Hi this issue (<https://github.com/dart-lang/sdk/issues/37101>) was fixed in Dart 2.3.2. The pub client from that version shouldn't have that problem. Thank you for your patience.
